app/commands/contributor_team/membership/destroy.rb

```diff
@@ -0,0 +1,21 @@
+class ContributorTeam::Membership
+  class Destroy
+    include Mandate
+
+    initialize_with :user, :team
+
+    def call
+      ContributorTeam::Membership.where(user: user, team: team).destroy_all
+
+      user.update(roles: user.roles - [team.role_for_members]) unless remains_in_team_role?
+
+      team.github_team.remove_membership(user.github_username)
+    end
+
+    private
+    memoize
+    def remains_in_team_role?
+      user.teams.any? { |t| t.role_for_members == team.role_for_members }
```

Comment:
```suggestion
      user.teams.pluck(:role_for_members).include?(team.role_for_members }
```

That will reduce object initialization and data transfer for someone like you who is in lots of teams, etc. (Code untested)